parking_permits_app/models.py

```diff
@@ -0,0 +1,386 @@
+import uuid
+
+from django.conf import settings
+from django.contrib.gis.db import models
+from django.utils.translation import ugettext_lazy as _
+
+
+# TODO: Some of these fields should come directly from Helsinki profile User-model.
+#  Check how to combine this model with Helsinki profile User-model.
+class Customer(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    first_name = models.CharField(_("First name"), max_length=32)
+    last_name = models.CharField(_("Last name"), max_length=32)
+    primary_address = models.ForeignKey(
+        "Address",
+        verbose_name=_("Primary address"),
+        on_delete=models.PROTECT,
+        related_name="customers_primary_address",
+    )
+    other_address = models.ForeignKey(
+        "Address",
+        verbose_name=_("Other address"),
+        on_delete=models.PROTECT,
+        related_name="customers_other_address",
+        blank=True,
+        null=True,
+    )
+    email = models.CharField(_("Email"), max_length=128, blank=True, null=True)
+    phone_number = models.CharField(
+        _("Phone number"), max_length=32, blank=True, null=True
+    )
+    parking_zone = models.ForeignKey(
+        "ParkingZone", verbose_name=_("Parking zone"), on_delete=models.PROTECT
+    )
+    terms_of_use_accepted = models.BooleanField(null=False, default=False)
+
+    class Meta:
+        db_table = "customer"
+        verbose_name = _("Customer")
+        verbose_name_plural = _("Customers")
+
+    def __str__(self):
+        return "%s - %s %s" % (self.id, self.first_name, self.last_name)
+
+
+class ParkingZone(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    name = models.CharField(_("Name"), max_length=128, blank=False, null=False)
+    location = models.MultiPolygonField(
+        _("Area (2D)"), srid=settings.SRID, blank=False, null=False
+    )
+
+    class Meta:
+        db_table = "parking_zone"
+        verbose_name = _("Parking zone")
+        verbose_name_plural = _("Parking zones")
+
+    def __str__(self):
+        return "%s - %s" % (self.id, self.name)
+
+
+class VehicleType(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    name = models.CharField(_("Name"), max_length=32, blank=False, null=False)
```

Comment:
Shouldn't this be strictly limited to `bensin`, `diesel` and `bifuel` based on the diagram you have provided? If so, I guess it can be written in the same way as `CATEGORIES` on `Vehicle` and be passed as `choices` to this field.

Reply:
Good point and will be done also.